Upgrade docz to 1.0.4

hsunpei opened the pull request:
# Purpose

Upgrade `docz` to `1.0.4` to increase the build speed when developing the project.

Some changes listed below are also included in this upgrade because type-checking issues emerged after the upgrade.

# Changes

- Upgrade `docz` to `1.0.4` to increase the build speed when developing the project.
- Prevent Jest tests from being checked by `ForkTSCheckerWebpackPlugin` of `docz` when developing
- Add `useWorkspaces` to allow `lerna bootstrap` to run
- Fix the typing problems raised by `@types/styled-components` and `react-spring`

# Screenshots

![Screen Shot 2019-04-29 at 6 27 07 PM](https://user-images.githubusercontent.com/1139698/56890589-74529880-6aac-11e9-8da2-6653aaeb229b.png)

Note that some styles such as the default fonts are subject to the changes made in the upgrade of `docz`.


# Risks

`<PropsTable>` is replaced with `<Props>` in the `docz` upgrade.

However, some of the property tables might not be able to show (such as `<Props>` in `<BarChart>` and `<LineChart>`) and have to wait for the fixes of `docz`.

Ref: https://github.com/pedronauck/docz/issues/777

hsunpei committed:
Upgrade docz to the latest version

hsunpei committed:
Replace PropsTable with Props for the docz upgrade

hsunpei committed:
Fix the issue that @types/styled-components importing react-native types

Ref: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/33311

hsunpei committed:
Add useWorkspaces to allow lerna bootstrap to run

hsunpei committed:
Fix the useSpring typings problem by upgrading react-spring

Ref: https://github.com/react-spring/react-spring/issues/653

hsunpei committed:
Upgrade fork-ts-checker-webpack-plugin and add jest types

hsunpei committed:
Prevent tests from being checked by ForkTSCheckerWebpackPlugin of docz

hsunpei committed:
Fix the style of Playground after the docz upgrade

hsunpei committed:
Add CHANGELOG for docz's upgrade

zhusee2 reviewed (approved):
Looks great in general 💯

hsunpei committed:
Temporarily freeze the version of @types/styled-components to 4.1.8

Temporarily freeze the version of @types/styled-components to 4.1.8 to prevent it from importing @types/react-native

Ref: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/34882
